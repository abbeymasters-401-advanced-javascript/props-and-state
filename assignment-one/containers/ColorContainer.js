import React, { Component } from 'react';
import ColorPicker from '../../components/colors/ColorPickercker';
import ColorDisplay from '../../components/colors/ColorDisplayplay';


export default class ColorContainer extends Component {
  state = {
    backgroundColor: 'black'
  };

  selectColor = backgroundColor => {
    this.setState({ backgroundColor });
  }

  render() {
    const colors = ['red', 'green', 'black', 'yellow'];
    const { backgroundColor } = this.state;

    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} />
        <ColorPicker
          selectColor={this.selectColor}
          colors={colors} />
      </>
    );
  }
}


