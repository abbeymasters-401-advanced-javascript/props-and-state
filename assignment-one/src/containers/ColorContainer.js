import React, { Component } from 'react';
import ColorPicker from '../components/colors/ColorPicker';
import ColorDisplay from '../components/colors/ColorDisplay';


export default class ColorContainer extends Component {
  state = {
    backgroundColor: 'black'
  };

  selectColor = backgroundColor => {
    this.setState({ backgroundColor });
  }

  render() {
    const colors = ['red', 'green', 'black', 'yellow'];
    const { backgroundColor } = this.state;

    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} />
        <ColorPicker
          selectColor={this.selectColor}
          colors={colors} />
      </>
    );
  }
}


