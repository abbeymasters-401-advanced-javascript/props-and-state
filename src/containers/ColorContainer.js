import React, { Component } from 'react';

export default class ColorContainer extends Component {
  
  state = {
    backgroundColor: 'black'
  };
  
  selectColor = backgroundColor => {
    this.setState({ backgroundColor });
  }

  render(

  )
}


