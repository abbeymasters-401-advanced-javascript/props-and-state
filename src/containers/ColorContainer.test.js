import React from 'react';
import { shallow } from 'enzyme';
import ColorContainer from './ColorContainer';

describe('Color Container', () => {

  it('renders color container component', () => {
    const wrapper = shallow(<ColorContainer />);
    expect(wrapper).toMatchSnapshot();
  });

});
