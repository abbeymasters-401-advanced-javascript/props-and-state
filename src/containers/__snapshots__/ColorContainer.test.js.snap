// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Container renders color container component 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="black"
  />
  <ColorPicker
    colors={
      Array [
        "red",
        "green",
        "black",
        "yellow",
      ]
    }
    selectColor={[Function]}
  />
</Fragment>
`;
